workflow-templates/im-deploy-on-prem-website.yml

```diff
@@ -0,0 +1,233 @@
+
+      - run: |
+          echo "The maintenance window ID is: ${{ steps.open-window.outputs.maintenance-window-id }}"
+          deploy-the-code.sh
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: ${{ env.DOTNET_VERSION }}
+
+      # Uncomment this if the site hasn't been created before
+      # - name: Create Web Site
+      #   uses: im-open/iis-site-create@v1.0.0
+      #   with:
+      #     server: '${{ secrets.iis_server }}'
+      #     website-name: '${{env.WEBSITE_NAME}}'
+      #     app-pool-name: '${{ secrets.pool-name }}'
+      #     website-host-header: '${{env.WEBSITE_HOST_HEADER}}'
+      #     website-path: '${{env.DEPLOYMENT_FOLDER}}'
+      #     website-cert-path: '${{ env.WEBSITE_CERT_PATH}}'
+      #     website-cert-password: '${{ secrets.site_cert_password }}'
+      #     website-cert-friendly-name: '${{env.WEBSITE_CERT_FRIENDLY_NAME}}'
+      #     service-account-id: '${{secrets.iis_admin_user}}'
+      #     service-account-password: '${{secrets.iis_admin_password}}'
+      #     server-public-key: ${{ env.IIS_SERVER_CERT_PATH}}
+
+      # Do NOT upload the DEPLOY_ZIP as an artifact after you've done this!
+      # DEPLOY_ZIP will contain all of your sensitive values!
+      # If an artifact is uploaded, only upload ARTIFACT_ZIP
+      - name: Update appsettings.json with Environment Specific Values
+        uses: im-open/add-or-update-json-properties@v1.0.1
+        with:
+          path-to-json-file: '${{ env.PROJECT_ROOT }}/appsettings.json'
+          # TODO:  add any appsettings.json items that need to be updated for the environment.  Format:
+          # {"SecretName": "SecretValues"},
+          # {"Nested.SecretName": "${{secrets.NESTED_SECRET}}"},
+          # {"Auth0.ClientId": "${{ secrets.AUTH0_CLIENT_ID }}"}
+          properties-to-update-or-add: |
+            [
+            ]
+
+      # TODO:  If config transformation is needed for different environments include environment name
+      # For example to perform transformations on a web.config using the web.production.config file:
+      # dotnet publish --configuration Release /p:EnvironmentName=Production
+      - name: Build and Publish
+        working-directory: ${{ env.PROJECT_ROOT }}
+        run: |
+          dotnet build --configuration Release
+          dotnet publish -c Release -o ./published_app --no-restore
+
+      - name: Zip the published site for faster deployment and uploads
+        working-directory: ${{ env.PROJECT_ROOT }}
+        # TODO:  If you plan on uploading artifacts, add this line to the run:
+        # (cd published_app && zip -r ../${{env.ARTIFACT_ZIP}} . -x */appsettings*.json appsettings*.json)
+        run: |
+          (cd published_app && zip -r ../${{env.DEPLOY_ZIP}} .)
+
+      - name: Stop App Pool
+        id: app-pool-stop
+        if: always()
```

Comment:
I'm curious why we want to always do this. I can see always wanting to do the start step in case the workflow was cancelled after this step, but I'm not sure why we'd need to do this if the workflow was cancelled before this step. Was this just how Octopus worked or something?